tests/e2e/vsphere.go

```diff
@@ -1077,3 +1076,47 @@ func (vs *vSphere) verifyDatastoreMatch(volumeID string, dsUrl string) bool {
 	framework.Logf("dsUrl from QueryCNSVolumeWithResult: %s, expected: %s", queryResult.Volumes[0].DatastoreUrl, dsUrl)
 	return queryResult.Volumes[0].DatastoreUrl == dsUrl
 }
+
+// cnsRelocateVolume relocates volume from one datastore to another using CNS relocate volume API
+func (vs *vSphere) cnsRelocateVolume(e2eVSphere vSphere, ctx context.Context, fcdID string,
+	dsRefDest vim25types.ManagedObjectReference) error {
+	relocateSpec := cnstypes.NewCnsBlockVolumeRelocateSpec(fcdID, dsRefDest)
+	var baseCnsVolumeRelocateSpecList []cnstypes.BaseCnsVolumeRelocateSpec
+	baseCnsVolumeRelocateSpecList = append(baseCnsVolumeRelocateSpecList, relocateSpec)
+	req := cnstypes.CnsRelocateVolume{
+		This:          cnsVolumeManagerInstance,
+		RelocateSpecs: baseCnsVolumeRelocateSpecList,
+	}
+
+	cnsClient, err := newCnsClient(ctx, vs.Client.Client)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	res, err := cnsmethods.CnsRelocateVolume(ctx, cnsClient, &req)
+	if err != nil {
+		return err
+	}
+	task, err := object.NewTask(e2eVSphere.Client.Client, res.Returnval), nil
+	taskInfo, err := cns.GetTaskInfo(ctx, task)
+	if err != nil {
+		return err
+	}
+	_, err = cns.GetTaskResult(ctx, taskInfo)
```

Comment:
we should wait for task to complete here and then check the task status

Reply:
Done